### ISSUE TITLE
Preview states of registry

### ISSUE THREAD

**Issue by kamilkisiela** [kamilkisiela/graphql-hive#2437]
A temporary state (a preview) with TTL of 14 days or something to fetch the schemas and supergraph from.
Something that is purely about the registry (CDN part) without schema checks and observability stuff.

**Comment by n1ru4l** [comment]
dumping some information form the internal chat:

so from the information we have, it feels like either a flagged publish

```
hive schema:publish --preview
```
or a new type of command

```
hive schema:preview
```

In terms of the CDN, we could use [the `x-amz-expiration`  header ](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_ResponseSyntaxhttps://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_ResponseSyntax) which is also supported by r2, for automatically deleting the resources after some time.

**Comment by erikwrede** [comment]
Tuning in here to add some points from the chat with @kamilkisiela:

- there should be a way to tag the previews with either a revision or some other name so the preview environment can be configured to use the correct preview / the preview is automatically assigned to the correct preview environment by tag
- multiple previews for different schema changes should be possible at the same time

for the use case I have in mind, a flagged publish would work just fine

**Comment by kamilkisiela** [comment]
Ability to publish schema multiple times to the same preview (based on id?)

**Comment by yukukotani** [comment]
Is this intended for something like previewing the schema of PRs?

**Comment by dotansimha** [comment]
> Is this intended for something like previewing the schema of PRs?

Yes, this way you can get the complete artifact from the CDN, without performing the `publish` action and effect the registry.

**Comment by comatory** [comment]
For me, the use case is explained in #4089 , I'd like to create CI automation which will run type-check with PR-based schema changes in front-end code.

**Comment by jenskm** [comment]
Hi! 

Any updates on this issue? We have a similar use case as @comatory, where we are looking to enhance our CI pipeline. The purpose is to verify that schema changes proposed in a merge/pull request are compatible with our frontend code. 

Ideally something like the `schema:fetch` command that would support the action id (commit sha) that has been used in a `schema:check`.
